File: AlgoExpert/Arrays/Easy/Sorted squared array.java
import java.util.*;

class Program {

  public int[] sortedSquaredArray(int[] array) {
    // Write your code here.
		int[] sortedSquares = new int[array.length];
		
		for(int i = 0; i < array.length; i++){ // log(n)
			sortedSquares[i] = array[i] * array[i];
		}
		
		Arrays.sort(sortedSquares); // n log(n)
		return sortedSquares;
  }
}

File: Python/AlgoExpert/Arrays/Easy/sorted_squared_array.py
def sortedSquaredArray(array):
    # Write your code here.
    sorted_squares= [0 for x in array]
    start, end = 0, len(array) - 1

    for i in range(len(array) - 1, -1, -1):
        sq1 = array[start] ** 2
        sq2 = array[end] ** 2
        if (sq1 > sq2):
            sorted_squares[i] = sq1
            start += 1
        else:
            sorted_squares[i] = sq2
            end -= 1
    return sorted_squares
